extractor.py

```diff
@@ -1,146 +1,186 @@
 #!/usr/bin/env python3
-
 from argparse import ArgumentParser
+from json import JSONDecodeError
+import config
+import json
 import mimetypes
 import os
 
-DEBUG = False
-
-class Program:
-    _tar_type = 'application/x-tar'
-    _gzip_type = 'gzip'
-    _zip_type = 'application/zip'
-    _7z_type = 'application/x-7z-compressed'
-
-    _tar_ext = '.tar'
-    _targz_ext = '.tar.gz'
-    _gzip_ext = '.gz'
-    _zip_ext = '.zip'
-    _7z_ext = '.7z'
-
-    _tar_prog = 'tar'
-    _targz_prog = 'tar'
-    _gzip_prog = 'guzip'
-    _zip_prog = 'unzip'
-    _7z_prog = '7z'
-    
-    _tar_attr = 'xvf'
-    _targz_attr = 'xzvf'
-    _gzip_attr = ''
-    _zip_attr = ''
-    _7z_attr = 'x'
-
-    exts = {
-            _tar_type: _tar_ext,
-            _gzip_type: _gzip_ext,
-            _zip_type: _zip_ext,
-            _7z_type: _7z_ext,
-    }
-
-    progs = {
-            _tar_ext: _tar_prog,
-            _targz_ext: _targz_prog,
-            _gzip_ext: _gzip_prog,
-            _zip_ext: _zip_prog,
-            _7z_ext: _7z_prog,
-    }
+DEBUG = True
+
+_attrs_str = 'attrs'
+_configs_str = 'configs'
+_encoding_str = 'encoding'
+_ext_str = 'ext'
+_result_str = 'result'
+_message_str = 'message'
+_error_str = 'error'
+_error_code_str = 'error_code'
+_prog_str = 'prog'
+
+def _logd(*args, sep=' ', end='\n', file=None):
+    if DEBUG:
+        print(args, sep, end, file)
```

Comment:
I think this could be defined on a separate file/class and used here and on another classes of the script.. something like Logger.logd(...)